src/sentry/api/serializers/models/user.py

```diff
@@ -250,22 +264,24 @@ def serialize(
     ) -> DetailedUserSerializerResponse:
         d = cast(DetailedUserSerializerResponse, super().serialize(obj, attrs, user))
 
-        # XXX(dcramer): we don't use is_active_superuser here as we simply
-        # want to tell the UI that we're an authenticated superuser, and
-        # for requests that require an *active* session, they should prompt
-        # on-demand. This ensures things like links to the Sentry admin can
-        # still easily be rendered.
-        d["isSuperuser"] = obj.is_superuser
-        d["permissions"] = [up.permission for up in attrs["permissions"]]
-        d["authenticators"] = [
-            {
-                "id": str(a.id),
-                "type": a.interface.interface_id,
-                "name": str(a.interface.name),
-                "dateCreated": a.created_at,
-                "dateUsed": a.last_used_at,
-            }
-            for a in attrs["authenticators"]
-        ]
-        d["canReset2fa"] = attrs["canReset2fa"]
+        # safety check to never return this informatoin if the acting user is not 1) this user, 2) an admin
+        if user.id == obj.id or user.is_superuser:
+            # XXX(dcramer): we don't use is_active_superuser here as we simply
+            # want to tell the UI that we're an authenticated superuser, and
+            # for requests that require an *active* session, they should prompt
+            # on-demand. This ensures things like links to the Sentry admin can
+            # still easily be rendered.
+            d["isSuperuser"] = obj.is_superuser
+            d["permissions"] = sorted(attrs["permissions"])
+            d["authenticators"] = [
```

Comment:
I agree that we probably shouldn't be exposing these for all users, but isn't removing these properties from most API response data a breaking change?

This [endpoint is documented](https://docs.sentry.io/api/organizations/list-an-organizations-users/) and the properties being removed are in the example. We should at least update the API docs as well.

Reply:
It is ok to create breaking changes which could never have been used in the first place. Agree tho we need to update docs (tho I will have to figure out how that works, assumed it was automatic).

Reply:
Looks like the only concern is `isSuperuser` is doc'd. So we must've added these additional properties not realizing it was being used in this endpoint.

Reply:
f/u for ref: superuser is exposed 100% of the time. so this code was duplicated and no longer relevant here